### ISSUE TITLE
Consent required

### ISSUE THREAD

**Issue by magwas** [auth0/auth0-angular#416]
### Describe the problem

Acquiring consent is not documented. Documented workaround to not ask for consent is not working.
When I issue an API call, using the interceptor provided by the package, a network request is sent to auth0, and it returns "Consent Required".

The document https://auth0.com/docs/quickstart/spa/angular/interactive does not talk about consent, so I do not know how to acquire it.

I have made the settings detailed in this doc: https://stackoverflow.com/questions/49091827/auth0-asks-for-consent-to-access-tenent-when-logging-in , but the error persists

### What was the expected behavior?

When I access an API resource I want to aquire consent of the user.

### Reproduction

Commit 041a874cba4c791c0a0af97a56f648b88b9722fe of https://github.com/kode-konveyor/angulartest/

### Environment

> Please provide the following:

- **Version of `auth0-angular` used:**
- **Which browsers have you tested in?**
- **Which version of Angular have you tested with?**
- **Other modules/plugins/libraries that might be involved:**

**Comment by frederikprijck** [comment]
Thanks for reaching out. Asking for consent is not a bug in our SDK.

What it means is that you are trying to use scopes for which your Auth0 application/API is configured to require user consent, meaning the user have to allow you to use those scopes. Without allowing that, we won't allow you to use those scopes, which isn't a bug but works as expected.

If you get a consent required error, you need to either do:

- Ensure u send the same scopes when logging in, this will allow Auth0 to ask the user for consent. If the user approves, u can continue. If not, you will stil let the consent required error and need to work around that as the user did not allow you to use the scopes.
- If you don't want that, disable consent for first party apps as per StackOverflow and ensure the application is registered as a first party app, more here: https://auth0.com/docs/get-started/applications/confidential-and-public-applications/first-party-and-third-party-applications

For the first solution, you need to ensure you pass the following when calling AuthModule.forRoot():

```ts
AuthModule.forRoot({
      domain: 'kode-konveyor.eu.auth0.com',
      clientId: 'OqUGGMvs9Ch8yitD3sf2lm6mN61MZqPw',
      authorizationParams: {
        redirect_uri: window.location.href,
        audience: 'https://test.kodekonveyor.com/angulartest',
        scope: 'read:current_user'
      },
      httpInterceptor: {
        allowedList: [
          {
            uri: 'https://test.kodekonveyor.com/angulartest/*',
            tokenOptions: {
              authorizationParams: {
                audience: 'https://test.kodekonveyor.com/angulartest',
                scope: 'read:current_user'
              }
            }
          }
        ]
      }
    })
```

Doing so should ask for consent when calling loginWithRedirect, and now when using the interceptor.

Let me know if that doesn't work.

**Comment by YannikG** [comment]
I faced the "consent required" exception in a freshly created project. I temporarily solved the problem by switching back to 1.11.0.

I might have missed some breaking changes in 2.0.0 regarding the http interceptor.

my working `app.module.ts` in version 1.11.0:
```
...
    AuthModule.forRoot({
      ...environment.auth,
      httpInterceptor: {
        allowedList : [  
          {
            uri: '/api/*',
            tokenOptions: {
              audience: environment.auth.audience,
              scope: environment.auth.scope
            }
          }
        ]
      }
    }),
...
```

**Comment by frederikprijck** [comment]
Thanks for reaching out.

What does your config looks like for V2 that is giving you a consent required error? Can u include everything, so also environment.auth as that's getting spread into the module's config.

Based on what i see, both audience and scope, currently inside tokenOptions, should go inside authorizationParams, inside tokenOptions to work with V2.

But i assume environment.auth also needs to change.

**Comment by YannikG** [comment]
Thanks for your response.

my environment currently looks like this. I haven't changed it while using v2.
```
export const environment = {
  production: false,
  auth: {
    domain: "" + auth_config.domain,
    clientId: "" + auth_config.clientId,
    redirectUri: window.location.origin,
    audience: "" + auth_config.audience,
    scope: "" + auth_config.scope
  }
};
```
I moved in v2 the audience and scope from tokenOptions into tokenOptions.authorizationParams. but got the consent required error.
Only when removing authorizationParams completely, I got no error. Unfortunately, my API then complaints regarding missing audience...

I have to check tomorrow If I'm able to reproduce it.

And before I forget it: the quickstart and documentations on Auth0 websites refer to v1.X.

**Comment by frederikprijck** [comment]
Thanks,

Both audience and scope need to go in yet another authorizationParams in the environment.auth. So does redirectUri, but it needs to be renamed to redirect_uri. Please see the [migration guide.](https://github.com/auth0/auth0-angular/blob/master/MIGRATION_GUIDE.md)

```
export const environment = {
  production: false,
  auth: {
    domain: "" + auth_config.domain,
    clientId: "" + auth_config.clientId,
    authorizationParams: {
      redirect_uri: window.location.origin,
      audience: "" + auth_config.audience,
      scope: "" + auth_config.scope
    }
  }
};
```

Can you elaborate on what docs and quickstart you refer to?

**Comment by magwas** [comment]
Note that the bug report was about the behaviour not/badly documented.
Not about not working as intended, but about that it is hard to figure out how it is intended to work.

**Comment by YannikG** [comment]
@magwas yes, I just wanted to let the developer know regarding the old documentations on the official website.

I might stumbled across an old [documentation](https://auth0.com/docs/libraries/auth0-angular-spa) but I also just found [this](https://auth0.com/docs/quickstart/spa/angular/02-calling-an-api) quickstart with correct information.

Regarding the issue I faced:
I have forgotten to change the environment as @frederikprijck mentioned. I try to confirm it tomorrow.

Thanks for your help :)
